src/tradelayer/dex.cpp

```diff
@@ -464,38 +460,29 @@ int DEx_acceptDestroy(const std::string& addressBuyer, const std::string& addres
     int rc = DEX_ERROR_ACCEPT -20;
     CMPOffer* p_offer = DEx_getOffer(addressSeller, propertyid);
     CMPAccept* p_accept = DEx_getAccept(addressSeller, propertyid, addressBuyer);
-    bool fReturnToMoney = true;
+    bool fReturnToMoney;
 
     if (!p_accept) return rc; // sanity check
 
     // if the offer is gone, ACCEPT_RESERVE should go back to seller's BALANCE
     // otherwise move the previously accepted amount back to SELLOFFER_RESERVE
     if (!p_offer) {
         fReturnToMoney = true;
-    } else {
-        PrintToLog("%s: finalize trade [offer=%s, accept=%s]\n", __func__,
-                        p_offer->getHash().GetHex(), p_accept->getHash().GetHex());
-
-        // offer exists, determine whether it's the original offer or some random new one
-        if (p_offer->getHash() == p_accept->getHash()) {
-            // same offer, return to SELLOFFER_RESERVE
-            fReturnToMoney = false;
-        } else {
-            // old offer is gone !
-            fReturnToMoney = true;
-        }
     }
 
+    PrintToLog("%s: finalize trade [offer=%s, accept=%s]\n", __func__, p_offer->getHash().GetHex(), p_accept->getHash().GetHex());
+
+    // offer exists, determine whether it's the original offer or some random new one
+    fReturnToMoney = (p_offer->getHash() == p_accept->getHash()) ? false : true;
+
     const int64_t amountRemaining = p_accept->getAcceptAmountRemaining();
 
+
     if (amountRemaining > 0) {
-        if (fReturnToMoney) {
-            assert(update_tally_map(addressSeller, propertyid, -amountRemaining, ACCEPT_RESERVE));
-            assert(update_tally_map(addressSeller, propertyid, amountRemaining, BALANCE));
-        } else {
-            assert(update_tally_map(addressSeller, propertyid, -amountRemaining, ACCEPT_RESERVE));
-            assert(update_tally_map(addressSeller, propertyid, amountRemaining, SELLOFFER_RESERVE));
-        }
+        TallyType typ = (fReturnToMoney) ? BALANCE : SELLOFFER_RESERVE;
+        assert(update_tally_map(addressSeller, propertyid, -amountRemaining, ACCEPT_RESERVE));
+        assert(update_tally_map(addressSeller, propertyid, amountRemaining, typ));
+
```

Comment:
Compita, para que es este nuevo "typ"?

Reply:
@santos177

Reply:
It's just only to not write the functions assert twice.